### ISSUE TITLE
Firefly Knulli update doesn’t allow TMNT full screen windowed  mode anymore

### ISSUE THREAD

**Issue by PapaBear7856** [knulli-cfw/distribution#227]
### Knulli build version

Firefly

### Your architecture

RG35XXSP

### Issue description

Previous Knulli version allowed full screen / windowed mode with Portmaster TMNT Shredders Revenge. Knulli Firefly disables this functionality even when changing display mode within the game

### Detailed reproduction steps

Install and updated Portmaster and TMNT to latest version and Knulli Firefly to latest version and unable to change display mode now

### Details of any modifications you have made to Knulli.

N/A

**Comment by dopazz** [comment]
The game is 16:9 and was improperly cropped to 4:3 previously.  The current behavior is the expected behavior.

**Comment by PapaBear7856** [comment]
True, the game is 16:9 but there are display options within the game’s actual settings that allow the original 16:9 or a windowed 4:3 display mode. Those options work correctly and allow you to switch between them on Knulli July version but do not function on Knulli Dec/Firefly.

**Comment by dopazz** [comment]
No, they worked incorrectly in the July release.  The current behavior is the expected behavior.

**Comment by PapaBear7856** [comment]
I don’t think you understand what I mean. In July version, there were options inside the game itself called Display Mode. It allowed you to CHOOSE. And each choice resulted in a functional difference. You could pick the widescreen mode and when chose that then it worked as widescreen mode with black bars. You could also choose in Display Mode the Windowed mode option and that correctly switched too. All of the display mode options worked and functioned correctly based on your CHOICE. On Knulli Firefly, no matter what mode you choose inside the game, it presents the same result and nothing functionally changes. This is a game option based on personal preference and for some reason that choice option is currently non-functional on current build.

**Comment by dopazz** [comment]
I understand that you want 4:3 back, but the game is 16:9.  It is working correctly now; it was never meant to be presented or played in 4:3.  The options you describe in the menu deal with how the game interfaces with the display and are not meant to change the aspect ratio of the content.

**Comment by PapaBear7856** [comment]
The game developers offered the option of Display Mode switching between widescreen and windowed modes. It was coded by them to offer this like a menu at a restaurant that used to give you exactly what you ordered (hotdog or pizza). Now the game still gives you menu options (hotdog or pizza) to change it but all you get is hot dogs no matter what you order, that wouldn’t be expected behavior at a restaurant since they made the menu items and are offering the customer choice. If this is “expected behavior” to only offer widescreen display modes then why did the developers of the game code it to be a choice selection? That aspect of the game is currently non-functional.

**Comment by dopazz** [comment]
I don't think you understand that option in the menu.  The word "widescreen" is not in any of the choices because this option does not change the aspect ratio of the game content.  The Display Mode options are: Windowed, Borderless Fullscreen, and Exclusive Fullscreen.  In Knulli we're not running a GUI window manager so these options aren't really relevant, everything launched from EmulationStation (emulators, ports, tools) gets fullscreen access to the display.

![knulli-screenshot_640x480_20241222_06h36m31s](https://github.com/user-attachments/assets/766381be-06a3-461c-b67b-81ad62b12ff5)

Try running the game on your PC through Steam or GOG, whichever you own.  You'll see that even when you run the game in an environment that can handle windowing the only resolution options you get are 16:9.

![image](https://github.com/user-attachments/assets/f08f0dc8-6d16-461c-acbd-bdeecaea7f95)


It's all 🍕with the Turtles, no 🌭
